pkg/pgmodel/pgx.go

```diff
@@ -1021,6 +1025,81 @@ func (q *pgxQuerier) LabelValues(labelName string) ([]string, error) {
 	return labelValues, nil
 }
 
+const GetLabelsSQL = "SELECT (key_value_array($1::int[])).*"
```

Comment:
(don't feel strongly) should this cache stuff be broken out into it's own .go file?

Reply:
Not yet, but it should later: we should split out the caching layer from the talks-to-database layer, with only that final layer in pgx and a well-defined go api between them. This would make our tests _much_ nicer as we can tests value-injection from the database using the go api, and only test SQL output from the pgx layer.